subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultArtifactIdentifier.java

```diff
@@ -81,23 +84,19 @@ public boolean equals(Object o) {
 
         DefaultArtifactIdentifier that = (DefaultArtifactIdentifier) o;
 
-        if (classifier != null ? !classifier.equals(that.classifier) : that.classifier != null) {
-            return false;
-        }
-        if (extension != null ? !extension.equals(that.extension) : that.extension != null) {
+        if (!Objects.equals(classifier, that.classifier)) {
```

Comment:
:x: We should not introduce Guava `Objects`. While convenient, it has the drawback of creating arrays for every comparison, which turns out to create a lot of garbage collection.